app/addons/dataimporter/components.react.jsx

```diff
@@ -0,0 +1,797 @@
+        </div>
+      );
+    },
+
+    boxIsDraggingOver: function () {
+      return (
+        <div className="dropzone dragging-file-in-background"
+          onDragOver={this.dragOver}
+          onDragLeave={this.endDragOver}
+          onDrop={this.drop}>
+          <div className="dropzone-msg draggingover">
+            <span className="fonticon icon-file-text-alt">
+            </span>
+            <span className="loading-msg">
+              Drop your file.
+            </span>
+          </div>
+        </div>
+      );
+    },
+
+    boxIsLoading: function () {
+      var loadingBoxBigFileMessage = this.props.isBigFile ?
+        this.loadingBoxBigFileMessage() : '';
+
+      return (
+        <div className={"dropzone loading-background"}>
+          <div className="dropzone-msg loading">
+            Loading...
+            {loadingBoxBigFileMessage}
+            <Components.LoadLines />
+          </div>
+        </div>
+      );
+    },
+
+    boxShowLimitInfo: function () {
+      return (
+         <div className="dropzone limit-info"
+          onDragOver={this.dragOver}
+          onDragLeave={this.endDragOver}
+          onDrop={this.drop}>
+          <div className="dropzone-msg">
+            <p>150 MB filesize limit</p>
+            <p>Only .csv files will import correctly</p>
+            <p>Fine grained import options are only for files under 200KB</p>
+          </div>
+          {this.fileLimitLink("Close")}
+        </div>
+      );
+    },
+
+    render: function () {
+      var box = this.defaultBox();
+
+      if (this.state.draggingOver) {
+        box = this.boxIsDraggingOver();
+      } else if (this.state.loading) {
+        box = this.boxIsLoading();
+      } else if (this.state.showLimitInfo) {
+        box = this.boxShowLimitInfo();
+      }
+      return box;
+    }
+  });
+
+  var DataImporterPreviewData = React.createClass({
+    getInitialState: function () {
+      return {
+        left: 0
+      };
+    },
+    bigFilePreviewWarning: function () {
+      var rowShown = this.props.rowShown,
+          totalRows = this.props.rowsTotal;
+
+      return (
+        <div className="top-row">
+          <div className="big-file-info-message">
+            <p className="big-file-preview-limit-info-message">
+              Because of the size of this file, this preview only shows the
+              first {rowShown} rows, out of {totalRows} rows total.
+              However, if you choose to load the data into a database, the
+              entirety of the file (all {totalRows} rows) will be imported.
+            </p>
+          </div>
+          {this.loadingIntoDB()}
+        </div>
+      );
+    },
+    componentDidMount: function () {
+      document.getElementById('preview-page')
+        .addEventListener('scroll', this.handleScroll);
+    },
+    componentWillUnmount: function () {
+      document.getElementById('preview-page')
+        .removeEventListener('scroll', this.handleScroll);
+    },
```

Comment:
no special reason for not using jquery. I just felt like it. 

this jquery rerendering part was a tough decision. It's expensive performance-wise, but when I spent a while on the css for this, and considered how it would look to people who have the scrolls always on. It looks weird.

It gets even more hairy when we put the component into its own parent element, then also the parent element won't move out of screen when we scroll down.  This is the UI I wanted, but the it's at the expense of performance, for now. There a few logic things I'd like to tackle, before I redo the CSS.  I'd like to redo that in another PR.